fs-storage/src/file_storage.rs

```diff
@@ -90,41 +85,53 @@ where
     V: Clone
         + serde::Serialize
         + serde::de::DeserializeOwned
-        + std::str::FromStr,
+        + std::str::FromStr
+        + Monoid<V>,
 {
     /// Set a key-value pair in the storage
     fn set(&mut self, key: K, value: V) {
         self.data.entries.insert(key, value);
-        self.timestamp = std::time::SystemTime::now();
-        self.write_fs()
-            .expect("Failed to write data to disk");
+        self.modified = std::time::SystemTime::now();
     }
 
     /// Remove a key-value pair from the storage given a key
     fn remove(&mut self, id: &K) -> Result<()> {
         self.data.entries.remove(id).ok_or_else(|| {
             ArklibError::Storage(self.label.clone(), "Key not found".to_owned())
         })?;
-        self.timestamp = std::time::SystemTime::now();
+        self.modified = std::time::SystemTime::now();
         self.write_fs()
             .expect("Failed to remove data from disk");
         Ok(())
     }
 
     /// Compare the timestamp of the storage file with the timestamp of the storage instance
-    /// to determine if the storage file has been updated.
-    fn is_storage_updated(&self) -> Result<bool> {
-        let file_timestamp = fs::metadata(&self.path)?.modified()?;
-        let file_time_secs = file_timestamp
-            .duration_since(SystemTime::UNIX_EPOCH)
-            .unwrap()
-            .as_secs();
-        let self_time_secs = self
-            .timestamp
-            .duration_since(SystemTime::UNIX_EPOCH)
-            .unwrap()
-            .as_secs();
-        Ok(file_time_secs > self_time_secs)
+    /// to determine if the storage file is up-to-date.
+    fn is_outdated(&self) -> Result<bool> {
+        match fs::metadata(&self.path) {
+            Ok(metadata) => {
+                let fs_modified = metadata
+                    .modified()?
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs();
```

Comment:
Probably a convenience function would be nice to have, to reduce code duplication now and in future.